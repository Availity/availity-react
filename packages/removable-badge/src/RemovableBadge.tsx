import React from 'react';
import { Badge, BadgeProps } from 'reactstrap';
import Icon from '@availity/icon';

export type RemovableBadgeProps = {
  children: string;
  value: string;
  onRemove: (value: string) => void;
} & BadgeProps;

const RemovableBadge = ({ children, value, onRemove, ...rest }: RemovableBadgeProps): JSX.Element => (
  <Badge data-testid="removable_badge" {...rest}>
    <Icon data-testid="removable_badge_remove" name="cancel" onClick={() => onRemove(value)} />
    {children}
  </Badge>
);

export default RemovableBadge;
