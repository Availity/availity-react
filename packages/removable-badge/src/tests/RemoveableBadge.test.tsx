import React from 'react';
import { render, waitFor, fireEvent } from '@testing-library/react';
import RemovableBadge from '../RemovableBadge';

describe('Removable Badge', () => {
  test('should render', async () => {
    const onRemove = jest.fn();
    const displayText = 'My Badge';
    const color = 'primary';

    const { container, getByTestId } = render(
      <RemovableBadge onRemove={onRemove} value="badgeValue" color={color}>
        {displayText}
      </RemovableBadge>
    );

    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();

    const badgeElement = await waitFor(() => getByTestId('removable_badge'));
    expect(badgeElement).not.toBeNull();
    expect(badgeElement.textContent).toBe(displayText);
    expect(badgeElement.className).toContain(color);
  });

  test('should call onRemove onClick', async () => {
    const onRemove = jest.fn();
    const displayText = 'My Badge';
    const color = 'primary';

    const { container, getByTestId } = render(
      <RemovableBadge onRemove={onRemove} value="badgeValue" color={color}>
        {displayText}
      </RemovableBadge>
    );

    expect(container).toBeDefined();

    const badgeElement = await waitFor(() => getByTestId('removable_badge'));
    expect(badgeElement).not.toBeNull();

    const removeIcon = await waitFor(() => getByTestId('removable_badge_remove'));
    expect(removeIcon).not.toBeNull();
    fireEvent.click(removeIcon);

    await waitFor(() => {
      expect(onRemove).toHaveBeenCalledTimes(1);
    });
  });
});
