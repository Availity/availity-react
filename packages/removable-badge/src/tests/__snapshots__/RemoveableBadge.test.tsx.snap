// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Removable Badge should render 1`] = `
<div>
  <span
    class="badge badge-primary"
    data-testid="removable_badge"
  >
    <i
      aria-hidden="true"
      class="icon icon-cancel"
      data-testid="removable_badge_remove"
      style="cursor: pointer;"
    />
    My Badge
  </span>
</div>
`;
